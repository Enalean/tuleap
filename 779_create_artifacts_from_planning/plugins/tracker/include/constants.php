<?php
/**
 * Copyright (c) Enalean, 2012. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

define('TRACKER_BASE_URL', '/plugins/tracker');
define('TRACKER_BASE_DIR', dirname(__FILE__));
define('TRACKER_EVENT_INCLUDE_CSS_FILE', 'tracker_event_include_css_file');

/**
  * The trackers from a project have been duplicated in another project
  *
  * Parameters:
  * 'tracker_mapping' => The mapping between source and target project trackers
  * 'group_id'        => The id of the target project
  *
  * No expected results
  */
define('TRACKER_EVENT_TRACKERS_DUPLICATED', 'tracker_event_trackers_duplicated');

?>
