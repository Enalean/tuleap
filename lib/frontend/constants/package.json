{
  "author": "Enalean Team",
  "name": "@tuleap/core-constants",
  "homepage": "https://tuleap.org",
  "license": "GPL-2.0-or-later",
  "private": true,
  "type": "module",
  "module": "./dist/core-constants.js",
  "main": "./dist/core-constants.umd.cjs",
  "types": "./src/main.ts",
  "exports": {
    ".": {
      "types": "./src/main.ts",
      "import": "./dist/core-constants.js",
      "require": "./dist/core-constants.umd.cjs"
    }
  },
  "devDependencies": {
    "@tuleap/build-system-configurator": "workspace:*"
  },
  "scripts": {
    "typecheck": "tsc --noEmit",
    "build": "vite build",
    "watch": "vite build --watch --mode development --minify false"
  }
}
