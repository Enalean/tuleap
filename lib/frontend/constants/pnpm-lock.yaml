lockfileVersion: '6.0'

settings:
  autoInstallPeers: false
  excludeLinksFromLockfile: false

dependencies:
  '@tuleap/locale':
    specifier: workspace:^
    version: link:../locale

devDependencies:
  '@tuleap/build-system-configurator':
    specifier: workspace:*
    version: link:../build-system-configurator
