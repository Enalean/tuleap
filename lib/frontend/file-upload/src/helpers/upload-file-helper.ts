/*
 *  Copyright (c) Enalean, 2024 - Present. All Rights Reserved.
 *
 *  This file is a part of Tuleap.
 *
 *  Tuleap is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  Tuleap is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { type DetailedError, Upload } from "tus-js-client";
import { computedProgress } from "./progress-computation-helper";
import { Option } from "@tuleap/option";
import type { FileUploadOptions, OngoingUpload, OnGoingUploadFile } from "../file-upload-options";

export function uploadFile(
    files: OnGoingUploadFile[],
    file: File,
    upload_href: string,
    onProgressCallback: FileUploadOptions["onProgressCallback"],
    uploaders: Array<Upload>,
): Promise<Option<OngoingUpload>> {
    let uploader: Upload | null;

    const removeUploaderFromList = (): void => {
        if (!uploader) {
            return;
        }

        const index = uploaders.findIndex((file_uploader) => file_uploader === uploader);
        if (index !== -1) {
            uploaders.splice(index, 1);
        }
    };

    return new Promise((resolve, reject): void => {
        uploader = new Upload(file, {
            uploadUrl: upload_href,
            metadata: {
                filename: file.name,
                filetype: file.type,
            },
            onProgress: (bytes_sent: number, bytes_total: number): void => {
                const progress = computedProgress(files, file.name, bytes_sent, bytes_total);
                onProgressCallback(file.name, progress);
            },
            onSuccess: (): void => {
                removeUploaderFromList();
                return resolve(Option.nothing<OngoingUpload>());
            },
            onError: (error: Error | DetailedError): void => {
                return reject(error);
            },
        });
        uploader.start();
        uploaders.push(uploader);
    }).then(() => {
        if (!uploader) {
            throw new Error("The uploader has not been initialized properly.");
        }
        return Option.fromValue({
            cancel: () => {
                uploader?.abort();
                removeUploaderFromList();
            },
        });
    });
}
