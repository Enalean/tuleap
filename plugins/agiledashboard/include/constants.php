<?php
/**
 * Copyright (c) Enalean, 2012. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

define('AGILEDASHBOARD_BASE_URL', '/plugins/agiledashboard');
define('AGILEDASHBOARD_BASE_DIR', dirname(__FILE__));

/**
 * Get the additional panes to display next to a milestone in the agiledashboard
 *
 * Parameters:
 * 'milestone' => The current Planning_Milestone
 *
 * Expected results
 * 'panes'     => Input/Output parameter, array of type AgileDashboard_Pane
 */
define('AGILEDASHBOARD_EVENT_ADDITIONAL_PANES_ON_MILESTONE', 'agiledashboard_event_additional_panes_on_milestone');

/**
 * Modify the redirect parameters when attempt to display a planning without specific Milestone selected
 *
 * Parameters:
 * 'milestone' => The most recent Planning_Milestone on which we are about to be redirected
 *
 * Expected results
 * 'redirect_parameters' => Input/Output parameter, array of 'key' => 'value'
 */
define('AGILEDASHBOARD_EVENT_MILESTONE_SELECTOR_REDIRECT', 'agiledashboard_event_milestone_selector_redirect');
?>
