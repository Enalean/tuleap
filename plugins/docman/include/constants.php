<?php
/**
 * Copyright (c) Enalean, 2014-Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

require_once('DocmanConstants.php');

define('PLUGIN_DOCMAN_BASE_DIR', dirname(__FILE__));

/**
 * Event raised when someone comments an approval table
 *
 * parameters =>
 *   'item'       => Docman_Item
 *   'version_nb' => Docman_Version
 *   'table'      => Docman_ApprovalTable
 *   'review'     => Docman_ApprovalReviewer
 */
define('PLUGIN_DOCMAN_EVENT_APPROVAL_TABLE_COMMENT', 'plugin_docman_approval_table_comment');



/**
 * Event raised when a new empty doc is created
 *
 * Parameters:
 *      'item' => Docman_Item
 */
define('PLUGIN_DOCMAN_EVENT_NEW_EMPTY', 'plugin_docman_event_new_empty');

/**
 * Event raised when a new link is created
 *
 * Parameters:
 *      'item' => Docman_Item
 */
define('PLUGIN_DOCMAN_EVENT_NEW_LINK', 'plugin_docman_event_new_link');

/**
 * Event raised when a new folder is created
 *
 * Parameters:
 *      'item' => Docman_Item
 */
define('PLUGIN_DOCMAN_EVENT_NEW_FOLDER', 'plugin_docman_event_new_folder');

/**
 * Event raised when a docman link item ihas new version
 *
 * Parameters:
 *      'item'    => Docman_Link
 *      'version' => Docman_LinkVersion
 */
define('PLUGIN_DOCMAN_EVENT_NEW_LINKVERSION', 'plugin_docman_event_new_linkVersion');

/**
 * Event raised when a new Docman wiki item is created
 *
 * Parameters:
 *      'item'      => Docman_Item
 *      'group_id'  => integer
 *      'wiki_page' => string
 */
define('PLUGIN_DOCMAN_EVENT_NEW_PHPWIKI_PAGE', 'plugin_docman_event_new_wikipage');

/**
 * Event raised when a new Docman embedded file/file  item is created
 *
 * Parameters:
 *      'item'            => Docman_Item,
 *      'version'         => Docman_Version,
 */
define('PLUGIN_DOCMAN_EVENT_NEW_FILE', 'plugin_docman_after_new_document');

/**
 * Event raised when a new Docman embedded file/file version is created
 *
 * Parameters:
 *      'item'            => Docman_Item,
 *      'version'         => Docman_Version,
 */
define('PLUGIN_DOCMAN_EVENT_NEW_FILE_VERSION', 'plugin_docman_event_new_version');

/**
 * Event raised when docman need to get a phpwiki page
 *
 * Parameters:
 *      'phpwiki_page_name'
 *      'project_id'
 *      'phpwiki_page'
 */
define('PLUGIN_DOCMAN_EVENT_GET_PHPWIKI_PAGE', 'plugin_docman_event_get_phpwiki_page');


define('PLUGIN_DOCMAN_APPROVAL_TABLE_DISABLED', 0);
define('PLUGIN_DOCMAN_APPROVAL_TABLE_ENABLED', 1);
define('PLUGIN_DOCMAN_APPROVAL_TABLE_CLOSED', 2);
define('PLUGIN_DOCMAN_APPROVAL_TABLE_DELETED', 3);

define('PLUGIN_DOCMAN_APPROVAL_STATE_NOTYET', 0);
define('PLUGIN_DOCMAN_APPROVAL_STATE_APPROVED', 1);
define('PLUGIN_DOCMAN_APPROVAL_STATE_REJECTED', 2);
define('PLUGIN_DOCMAN_APPROVAL_STATE_COMMENTED', 3);
define('PLUGIN_DOCMAN_APPROVAL_STATE_DECLINED', 4);

define('PLUGIN_DOCMAN_APPROVAL_NOTIF_DISABLED', 0);
define('PLUGIN_DOCMAN_APPROVAL_NOTIF_ALLATONCE', 1);
define('PLUGIN_DOCMAN_APPROVAL_NOTIF_SEQUENTIAL', 2);

define('PLUGIN_DOCMAN_METADATA_TYPE_DATE_LABEL', 'date');

define('PLUGIN_DOCMAN_METADATA_TYPE_TEXT', 1);
define('PLUGIN_DOCMAN_METADATA_TYPE_STRING', 6);
define('PLUGIN_DOCMAN_METADATA_TYPE_DATE', 4);
define('PLUGIN_DOCMAN_METADATA_TYPE_LIST', 5);

define('PLUGIN_DOCMAN_METADATA_UNUSED', 0);
define('PLUGIN_DOCMAN_METADATA_USED', 1);
