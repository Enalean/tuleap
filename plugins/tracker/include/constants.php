<?php
/**
 * Copyright (c) Enalean, 2012. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */
define('TRACKER_BASE_URL', '/plugins/tracker');
define('TRACKER_BASE_DIR', dirname(__FILE__));
define('TRACKER_EVENT_INCLUDE_CSS_FILE', 'tracker_event_include_css_file');

/**
 * The trackers from a project have been duplicated in another project
 *
 * Parameters:
 * 'tracker_mapping' => The mapping between source and target project trackers
 * 'group_id'        => The id of the target project
 *
 * No expected results
 */
define('TRACKER_EVENT_TRACKERS_DUPLICATED', 'tracker_event_trackers_duplicated');

/**
 * An artifact has just been created/updated. Redirect to a plugin specific url if needed.
 *
 * Parameters:
 * 'request'  => The initial request
 * 'artifact' => The involved artifact
 *
 * Either a redirection has been done or nothing has been done 
 * (in this case plugins/tracker will commpute the redirection)
 */
define('TRACKER_EVENT_REDIRECT_AFTER_ARTIFACT_CREATION_OR_UPDATE', 'tracker_event_redirect_after_artifact_creation_or_update');

/**
 * We build the form action for a new artifact. Let the plugin inject its own variable.
 *
 * Parameters:
 * 'request'          => The initial request
 * 'query_parameters' => The actual form action parameters
 *
 * No expected results than the query_parameters modified if needed 
 */
define('TRACKER_EVENT_BUILD_ARTIFACT_FORM_ACTION', 'tracker_event_build_artifact_form_action');

/**
 * Get the admin items to display in the admin menu of a tracker
 *
 * Parameters:
 * 'tracker' => The current tracker object
 * 'items'   => The items to display. Inject your own items in this array to 
 *            display them in the menu. Each item is an associated array. Eg:
 *            'item_key' => array(
 *              'url',          => // The url to the admin subpage
 *              'short_title',  => // The short title of the item (in toolbar)
 *              'title',        => // The title (on the main admin page)
 *              'description'   => // Displayed in main admin page
 *              'img'           => // The src of the icon. As of today, 48x48px
 *            )
 *
 * No expected results than the query_parameters modified if needed 
 */
define('TRACKER_EVENT_ADMIN_ITEMS', 'tracker_event_admin_items');

/**
 * Let someone process the request on a given tracker.
 *
 * Parameters:
 * 'tracker'               => The current tracker object
 * 'func'                  => The action requested by the user
 * 'layout'                => Tracker_IDisplayTrackerLayout
 * 'request'               => The request object
 * 'user'                  => The user who made the request
 * 'nothing_has_been_done' => flag default to true
 *
 * Expected results:
 *  The action is handled by the listener and the flag nothing_has_been_done 
 *  is set to false.
 *  Or nothing has been done (flag untouched) and the core continue with the 
 *  default processing (eg: display the tracker)
 */
define('TRACKER_EVENT_PROCESS', 'tracker_event_process');

/**
 * An artifact has just been (un)associated to another one
 *
 * Parameters:
 * 'artifact'             => The artifact which receive the (un)association
 * 'linked-artifact-id'   => The (previously) linked artifact id
 * 'request'              => The request
 * 'user'                 => The user who made the request
 * 'form_element_factory' => The FormElementFactory
 *
 * Expected results:
 *  No expected results
 */
define('TRACKER_EVENT_ARTIFACT_ASSOCIATION_EDITED', 'tracker_event_artifact_association_edited');

/**
 * Should we display a selector to choose the parent of an item during creation?
 * If so, which artifacts are possible parents for the created item?
 *
 * By default, we display the selector with open artifacts parents
 *
 * Parameters:
 * 'user'             => User    The current user
 * 'parent_tracker'   => Tracker The parent tracker
 *
 * Re
 * 'possible_parents' => array of Tracker_Artifact
 * 'label'            => string the label of the possible parents list
 * 'display_selector' => bool true if we can display the selector
 */
define('TRACKER_EVENT_ARTIFACT_PARENTS_SELECTOR', 'tracker_event_artifact_parents_selector');
?>
