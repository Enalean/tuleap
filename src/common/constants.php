<?php
/*
 * Copyright (c) Enalean, 2016-Present. All Rights Reserved.
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, If not, see <http://www.gnu.org/licenses/>
 */

/* Global constants are evil, please avoid them whenever possible.
 * Use class constants instead.
 */
define('CODENDI_PURIFIER_CONVERT_HTML', Codendi_HTMLPurifier::CONFIG_CONVERT_HTML);
define('CODENDI_PURIFIER_STRIP_HTML', Codendi_HTMLPurifier::CONFIG_STRIP_HTML);
define('CODENDI_PURIFIER_BASIC', Codendi_HTMLPurifier::CONFIG_BASIC);
define('CODENDI_PURIFIER_BASIC_NOBR', Codendi_HTMLPurifier::CONFIG_BASIC_NOBR);
define('CODENDI_PURIFIER_LIGHT', Codendi_HTMLPurifier::CONFIG_LIGHT);
define('CODENDI_PURIFIER_FULL', Codendi_HTMLPurifier::CONFIG_FULL);
define('CODENDI_PURIFIER_JS_QUOTE', Codendi_HTMLPurifier::CONFIG_JS_QUOTE);
define('CODENDI_PURIFIER_DISABLED', Codendi_HTMLPurifier::CONFIG_DISABLED);
