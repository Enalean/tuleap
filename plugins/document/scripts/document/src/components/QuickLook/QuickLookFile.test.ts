/*
 * Copyright (c) Enalean 2022 - Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 *
 */

import { shallowMount } from "@vue/test-utils";
import QuickLookFile from "./QuickLookFile.vue";
import { TYPE_FILE } from "../../constants";
import type { ItemFile, FileProperties } from "../../type";
import { getGlobalTestOptions } from "../../helpers/global-options-for-test";

describe("QuickLookFile", () => {
    it("renders quick look for file document with a CTA to download the file", () => {
        const item = {
            id: 42,
            title: "my document",
            file_properties: {
                file_name: "my file",
                file_type: "image/png",
                download_href: "/plugins/docman/download/119/42",
                open_href: "",
                file_size: 109768,
            } as FileProperties,
            type: TYPE_FILE,
        } as ItemFile;

        const wrapper = shallowMount(QuickLookFile, {
            props: { item: item },
            global: { ...getGlobalTestOptions({}) },
        });

        expect(wrapper.element).toMatchSnapshot();
    });
    it("renders quick look for file document with a CTA to open the file", () => {
        const item = {
            id: 42,
            title: "my document",
            file_properties: {
                file_name: "my file",
                file_type: "image/png",
                download_href: "/plugins/docman/download/119/42",
                open_href: "/path/to/open/119",
                file_size: 109768,
            } as FileProperties,
            type: TYPE_FILE,
        } as ItemFile;

        const wrapper = shallowMount(QuickLookFile, {
            props: { item: item },
            global: { ...getGlobalTestOptions({}) },
        });

        const cta = wrapper.find<HTMLAnchorElement>(
            "[data-test=document-quick-look-document-cta-open]",
        ).element;
        expect(cta.href).toContain("/path/to/open/119");
    });
});
