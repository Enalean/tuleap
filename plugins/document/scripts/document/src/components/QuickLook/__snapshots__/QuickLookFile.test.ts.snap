// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickLookFile renders quick look for file document with a CTA to download the file 1`] = `
<div
  class="document-quick-look-document-action"
>
  <download-button-stub
    class="tlp-button-primary tlp-button-small"
    item="[object Object]"
  />
  <drop-down-quick-look-stub
    item="[object Object]"
  />
</div>
`;
